docs/src/main/asciidoc/microprofile-graphql.adoc

```diff
@@ -0,0 +1,416 @@
+and enables powerful developer tools.
+
+**GraphQL** was originally developed by **Facebook** in 2012 and has been
+a open standard since 2015.
+
+GraphQL is not a replacement for REST API specification but merely a
+alternative. Unlike REST GraphQL API's have a ability to benefit the client by:
+
+Preventing Over and Underfetching::
+    REST are server-driven fixed data responses that can not be determined by
+    the client. Although the client does not require all the fields the client
+    must retrieve all the data hence `Overfetching`. A client may also require
+    multiple REST API calls according to the first call (HATEOAS) to retrieve
+    all the data that is required thereby `Unfetching`.
+
+API Evolution::
+    Since GraphQL API's returns data that are requested by the client adding additional
+    fields and capabilities to existing API will not create breaking changes to existing
+    clients.
+
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* less than 15 minutes
+* an IDE
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* Apache Maven {maven-version}
+
+== Architecture
+
+In this guide, we build a simple GraphQL application that exposes a GraphQL API
+at `/graphql`.
+
+This example was inspired by a popular graphQL API
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `microprofile-graphql-quickstart` {quickstarts-tree-url}/microprofile-graphql-quickstart[directory].
+
+== Creating the Maven Project
+
+First, we need a new project. Create a new project with the following command:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=microprofile-graphql-quickstart \
+    -DclassName="org.acme.microprofile.graphql.FilmResource" \
+    -Dextensions="smallrye-graphql"
+cd microprofile-graphql-quickstart
+----
+
+This command generates a Maven project, importing the `smallrye-graphql` extension which is an implementation of the MicroProfile GraphQL specification used in Quarkus.
+
+== Preparing an Application: GraphQL API
+
+In this section we will start creating the GraphQL API
+
+First, create a the following entities representing a hero from a galaxy far far away:
+
+[source,java]
+----
+package org.acme.microprofile.graphql;
+
+public class Film {
+    public String title;
+    public String episodeID;
+    public String openingCrawl;
+    public String director;
+    public String producers;
+    public LocalDate releaseDate;
+}
+
+public class Hero {
+
+    public String name;
+    public Double height;
+    public Integer mass;
+    public Boolean isDarkSide;
+    public LightSaber lightSaber;
```

Comment:
You might need getters and setters. I never tested public fields, but it might work.

Reply:
I can add setter and getters or I can add a comment line stating the setters and getters are omitted for brevity. However, that way make the documentation misleading.

I will try with public fields to see if it works and then will add setters and getters depending on that. 🙂

Reply:
I tried using public fields in the quickstart but got errors when omitting setter and getter methods. If I have understand correctly `setter` and `getter` methods must be included
in order to qualify as a field in GraphQL API.

Reply:
Yes, The spec is also not that clear on fields (https://github.com/eclipse/microprofile-graphql/issues/226) and we are discussing it. We can see if we can build support for that in SmallRye. I'll add a ticket.